lib/compat/wordpress-6.2/class-gutenberg-rest-templates-controller-6-2.php

```diff
@@ -0,0 +1,77 @@
+<?php
+/**
+ * REST API: Gutenberg_REST_Templates_Controller_6_2 class
+ *
+ * @package    Gutenberg
+ * @subpackage REST_API
+ */
+
+/**
+ * Base Templates REST API Controller.
+ */
+class Gutenberg_REST_Templates_Controller_6_2 extends Gutenberg_REST_Templates_Controller {
+
+	/**
+	 * Registers the controllers routes.
+	 *
+	 * @return void
+	 */
+	public function register_routes() {
+
+		register_rest_route(
+			$this->namespace,
+			'/' . $this->rest_base . '/lookup',
+			array(
+				array(
+					'methods'             => WP_REST_Server::READABLE,
+					'callback'            => array( $this, 'get_template_fallback' ),
+					'permission_callback' => array( $this, 'get_item_permissions_check' ),
+					'args'                => array(
+						'slug'            => array(
+							'description' => __( 'The slug of the template to get the fallback for', 'gutenberg' ),
+							'type'        => 'string',
+							'required'    => true,
+						),
+						'is_custom'       => array(
+							'description' => __( 'Indicates if a template is custom or part of the template hierarchy', 'gutenberg' ),
+							'type'        => 'boolean',
+						),
+						'template_prefix' => array(
+							'description' => __( 'The template prefix for the created template. This is used to extract the main template type ex. in `taxonomy-books` we extract the `taxonomy`', 'gutenberg' ),
+							'type'        => 'string',
+						),
+						'ignore_empty'    => array(
+							'description' => __( 'If true templates with empty content are ignored.', 'gutenberg' ),
+							'type'        => 'boolean',
+							'default'     => false,
+						),
+					),
+				),
+			)
+		);
+		parent::register_routes();
+		// Get fallback template content.
+	}
+
+	/**
+	 * Returns the fallback template for a given slug.
+	 *
+	 * @param WP_REST_Request $request The request instance.
+	 *
+	 * @return WP_REST_Response|WP_Error
+	 */
+	public function get_template_fallback( $request ) {
+		$hierarchy         = get_template_hierarchy( $request['slug'], $request['is_custom'], $request['template_prefix'] );
+		$fallback_template = null;
+		if ( true === $request['ignore_empty'] ) {
+			do {
+				$fallback_template = resolve_block_template( $request['slug'], $hierarchy, '' );
+				array_shift( $hierarchy );
+			} while ( ! empty( $hierarchy ) && empty( $fallback_template->content ) );
+		} else {
+			$fallback_template = resolve_block_template( $request['slug'], $hierarchy, '' );
+		}
+		$response = $this->prepare_item_for_response( $fallback_template, $request );
+		return rest_ensure_response( $response );
+	}
+}
```

Comment:
Maybe we can add a unit test for this endpoint?

Reply:
I added a unit test.